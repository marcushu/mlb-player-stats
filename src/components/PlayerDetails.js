import React from 'react';
import Table from 'react-bootstrap/Table';
import PropTypes from 'prop-types';

import './playerDetails.css';

function PlayerDetails(props) {
  const stats = props
  const pitching = ["era","np", "avg", "r","so", "h", "hr9","ip", "w", "gs", "sv"];
  const hittingStats = ["avg", "ab", "h", "r",  "hr", "rbi", "slg", "bb", "sb", "d", "t", "tb",];


  return (
    <div className="p-2 py-3">
      {stats.hitting.totalSize > 0 &&
        <>
          <span className="statHeader">Career Hitting</span><br />
          <Table size="sm">
            <thead className="hittingHeader">
              <tr>
                {hittingStats.map(x => (<th key={x + "h1"} className="hittingStatNames">{x}</th>))}
              </tr>
            </thead>
            <tbody>
              <tr className="tableRowFigs">
                {hittingStats.map(x => (<td key={x + "h2"} className="figures">{stats.hitting.row[x]}</td>))}
              </tr>
            </tbody>
          </Table>
        </>}

      {stats.seasonHitting.totalSize > 0 &&
        <>
          <span className="statHeader">Season Hitting</span><br />
          <Table size="sm">
            <thead className="hittingHeader">
              <tr>
                {hittingStats.map(x => (<th key={x + "sh1"}className="hittingStatNames">{x}</th>))}
              </tr>
            </thead>
            <tbody>
              <tr className="tableRowFigs">
                {hittingStats.map(x => (<td key={x + "sh2"} className="figures">{stats.seasonHitting.row[x]}</td>))}
              </tr>
            </tbody>
          </Table>
        </>}

      {stats.pitching.totalSize > 0 &&
        <>
          <span className="statHeader">Career Pitching</span><br />
          <Table size="sm">
            <thead className="pitchingHeader">
              <tr>
                {pitching.map(x => (<th key={x + "cp1"} className="pitchingStatNames">{x}</th> ))}
              </tr>
            </thead>
            <tbody>
              <tr className="tableRowFigs">
                {pitching.map(x => (<td  key={x + "cp2"} className="figures">{stats.pitching.row[x]}</td>))}
              </tr>
            </tbody>
          </Table>
        </>}

      {stats.seasonPitching.totalSize > 0 &&
        <>
         <span className="statHeader">Season Pitching</span><br />
          <Table size="sm">
            <thead className="pitchingHeader">
              <tr>
                {pitching.map(x => (<th key={x + "p2"} className="pitchingStatNames">{x}</th> ))}
              </tr>
            </thead>
            <tbody>
              <tr className="tableRowFigs">
                {pitching.map(x => (<td  key={x + "p2"} className="figures">{stats.seasonPitching.row[x]}</td>))}
              </tr>
            </tbody>
          </Table>
        </>}
    </div>
  );

}

PlayerDetails.propTypes = {
  seasonHitting: PropTypes.object.isRequired,
  hitting: PropTypes.object.isRequired,
  pitching: PropTypes.object.isRequired,
  seasonPitching: PropTypes.object.isRequired
}

export default PlayerDetails