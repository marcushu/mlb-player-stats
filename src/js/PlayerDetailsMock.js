const hitting = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row":
    {"hr":"165","gidp":"67","np":"13519","sac":"0","team_count":"4","sport_code":"mlb","hgnd":"272","rbi":"528","lob":"1331","babip":".305","gidp_opp":"544","tb":"1586","xbh":"362","sport_id":"1","bb":"240","avg":".273","slg":".497","ops":".824","hbp":"31","g":"834","d":"173","hfly":"168","so":"742","wo":"3","sport":"MLB","sf":"28","hpop":"6","tpa":"3490","h":"870","cs":"20","obp":".327","hldr":"424","t":"24","ao":"939","r":"475","go_ao":"0.78","sb":"43","ppa":"3.87","player_id":"493316","ab":"3191","ibb":"33","roe":"37","go":"734"}}

const seasonHitting = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}    

const pitching = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}

const seasonPitching = {
    "created":"2020-08-12T21:51:33",
    "totalSize":"1",
    "row": {}
}

export { hitting, seasonHitting, pitching, seasonPitching }