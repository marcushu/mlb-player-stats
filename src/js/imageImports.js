
import bravesImg from '../images/braves.gif';
import indiansImg from '../images/indians.gif';
import oriolesImg from '../images/orioles.gif';
import redSoxImg from '../images/redSox.gif';
import diamodBacks from '../images/diamondBacks.gif';
import sox from '../images/sox.gif';
import reds from '../images/reds.gif';
import cubs from '../images/cubs.gif';
import angels from '../images/angels.gif';
import as from '../images/as.gif';
import astros from '../images/astros.gif';
import blueJays from '../images/blueJays.gif';
import cardinals from '../images/cardinals.gif';
import dodgers from '../images/dodgers.gif';
import mariners from '../images/mariners.gif';
import marlins from '../images/marlins.gif';
import mets from '../images/mets.gif';
import padres from '../images/padres.gif';
import phillies from '../images/phillies.gif';
import pirates from '../images/pirates.gif';
import rangers from '../images/rangers.gif';
import rays from '../images/rays.gif';
import royals from '../images/royals.gif';
import yankees from '../images/yankees.gif';
import nationals from '../images/nationals.gif';
import giants from '../images/giants.gif';
import tigers from '../images/tigers.gif';
import brewers from '../images/brewers.gif'
import twins from '../images/twins.gif';
import rockies from '../images/rockies.gif';

export {
    rockies,
    nationals,
    giants,
    tigers,
    brewers,
    twins,
    bravesImg,
    indiansImg,
    oriolesImg,
    redSoxImg,
    diamodBacks,
    sox,
    reds,
    cubs,
    angels,
    as,
    astros,
    blueJays,
    cardinals,
    dodgers,
    mariners,
    marlins,
    mets,
    padres,
    phillies,
    pirates,
    rangers,
    rays,
    royals,
    yankees,
}


