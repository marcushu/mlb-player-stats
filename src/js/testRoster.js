const tstRoster = [
    {
      "position_txt": "1B",
      "weight": "245",
      "name_display_first_last": "Pete Alonso",
      "college": "Florida",
      "height_inches": "3",
      "starter_sw": "N",
      "jersey_number": "20",
      "end_date": "",
      "name_first": "Peter",
      "bats": "R",
      "team_code": "nyn",
      "height_feet": "6",
      "pro_debut_date": "2019-03-28T00:00:00",
      "status_code": "A",
      "primary_position": "3",
      "birth_date": "1994-12-07T00:00:00",
      "team_abbrev": "NYM",
      "throws": "R",
      "team_name": "New York Mets",
      "name_display_last_first": "Alonso, Pete",
      "name_use": "Pete",
      "player_id": "624413",
      "name_last": "Alonso",
      "team_id": "121",
      "start_date": "2019-03-28T00:00:00",
      "name_full": "Alonso, Pete"
    },
    {
      "position_txt": "P",
      "weight": "265",
      "name_display_first_last": "Dellin Betances",
      "college": "",
      "height_inches": "8",
      "starter_sw": "N",
      "jersey_number": "68",
      "end_date": "",
      "name_first": "Dellin",
      "bats": "R",
      "team_code": "nyn",
      "height_feet": "6",
      "pro_debut_date": "2011-09-22T00:00:00",
      "status_code": "A",
      "primary_position": "1",
      "birth_date": "1988-03-23T00:00:00",
      "team_abbrev": "NYM",
      "throws": "R",
      "team_name": "New York Mets",
      "name_display_last_first": "Betances, Dellin",
      "name_use": "Dellin",
      "player_id": "476454",
      "name_last": "Betances",
      "team_id": "121",
      "start_date": "2019-12-24T00:00:00",
      "name_full": "Betances, Dellin"
    }
]

export default tstRoster