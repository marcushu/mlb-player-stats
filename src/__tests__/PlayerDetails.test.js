import React from 'react';
import { render } from '@testing-library/react';
import PlayerDetails from '../components/PlayerDetails';

import { hitting, seasonHitting, pitching, seasonPitching } from '../js/PlayerDetailsMock';

describe('PlayerDetails', () => {

    it('renders', () => {
        const { asFragment } = render(
            <PlayerDetails
                hitting={hitting}
                pitching={pitching}
                seasonHitting={seasonHitting}
                seasonPitching={seasonPitching}
                />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('displays the details', () => {
        const { getByText } = render(
            <PlayerDetails
                hitting={hitting}
                pitching={pitching}
                seasonHitting={seasonHitting}
                seasonPitching={seasonPitching}
                />
        );

        expect(getByText('.273')).not.toBeEmpty();
    })
});