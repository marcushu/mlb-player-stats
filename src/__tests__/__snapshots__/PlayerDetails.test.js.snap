// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDetails renders 1`] = `
<DocumentFragment>
  <div
    class="p-2 py-3"
  >
    <span
      class="statHeader"
    >
      Career Hitting
    </span>
    <br />
    <table
      class="table table-sm"
    >
      <thead
        class="hittingHeader"
      >
        <tr>
          <th
            class="hittingStatNames"
          >
            avg
          </th>
          <th
            class="hittingStatNames"
          >
            ab
          </th>
          <th
            class="hittingStatNames"
          >
            h
          </th>
          <th
            class="hittingStatNames"
          >
            r
          </th>
          <th
            class="hittingStatNames"
          >
            hr
          </th>
          <th
            class="hittingStatNames"
          >
            rbi
          </th>
          <th
            class="hittingStatNames"
          >
            slg
          </th>
          <th
            class="hittingStatNames"
          >
            bb
          </th>
          <th
            class="hittingStatNames"
          >
            sb
          </th>
          <th
            class="hittingStatNames"
          >
            d
          </th>
          <th
            class="hittingStatNames"
          >
            t
          </th>
          <th
            class="hittingStatNames"
          >
            tb
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowFigs"
        >
          <td
            class="figures"
          >
            .273
          </td>
          <td
            class="figures"
          >
            3191
          </td>
          <td
            class="figures"
          >
            870
          </td>
          <td
            class="figures"
          >
            475
          </td>
          <td
            class="figures"
          >
            165
          </td>
          <td
            class="figures"
          >
            528
          </td>
          <td
            class="figures"
          >
            .497
          </td>
          <td
            class="figures"
          >
            240
          </td>
          <td
            class="figures"
          >
            43
          </td>
          <td
            class="figures"
          >
            173
          </td>
          <td
            class="figures"
          >
            24
          </td>
          <td
            class="figures"
          >
            1586
          </td>
        </tr>
      </tbody>
    </table>
    <span
      class="statHeader"
    >
      Season Hitting
    </span>
    <br />
    <table
      class="table table-sm"
    >
      <thead
        class="hittingHeader"
      >
        <tr>
          <th
            class="hittingStatNames"
          >
            avg
          </th>
          <th
            class="hittingStatNames"
          >
            ab
          </th>
          <th
            class="hittingStatNames"
          >
            h
          </th>
          <th
            class="hittingStatNames"
          >
            r
          </th>
          <th
            class="hittingStatNames"
          >
            hr
          </th>
          <th
            class="hittingStatNames"
          >
            rbi
          </th>
          <th
            class="hittingStatNames"
          >
            slg
          </th>
          <th
            class="hittingStatNames"
          >
            bb
          </th>
          <th
            class="hittingStatNames"
          >
            sb
          </th>
          <th
            class="hittingStatNames"
          >
            d
          </th>
          <th
            class="hittingStatNames"
          >
            t
          </th>
          <th
            class="hittingStatNames"
          >
            tb
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowFigs"
        >
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
        </tr>
      </tbody>
    </table>
    <span
      class="statHeader"
    >
      Career Pitching
    </span>
    <br />
    <table
      class="table table-sm"
    >
      <thead
        class="pitchingHeader"
      >
        <tr>
          <th
            class="pitchingStatNames"
          >
            era
          </th>
          <th
            class="pitchingStatNames"
          >
            np
          </th>
          <th
            class="pitchingStatNames"
          >
            avg
          </th>
          <th
            class="pitchingStatNames"
          >
            r
          </th>
          <th
            class="pitchingStatNames"
          >
            so
          </th>
          <th
            class="pitchingStatNames"
          >
            h
          </th>
          <th
            class="pitchingStatNames"
          >
            hr9
          </th>
          <th
            class="pitchingStatNames"
          >
            ip
          </th>
          <th
            class="pitchingStatNames"
          >
            w
          </th>
          <th
            class="pitchingStatNames"
          >
            gs
          </th>
          <th
            class="pitchingStatNames"
          >
            sv
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowFigs"
        >
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
        </tr>
      </tbody>
    </table>
    <span
      class="statHeader"
    >
      Season Pitching
    </span>
    <br />
    <table
      class="table table-sm"
    >
      <thead
        class="pitchingHeader"
      >
        <tr>
          <th
            class="pitchingStatNames"
          >
            era
          </th>
          <th
            class="pitchingStatNames"
          >
            np
          </th>
          <th
            class="pitchingStatNames"
          >
            avg
          </th>
          <th
            class="pitchingStatNames"
          >
            r
          </th>
          <th
            class="pitchingStatNames"
          >
            so
          </th>
          <th
            class="pitchingStatNames"
          >
            h
          </th>
          <th
            class="pitchingStatNames"
          >
            hr9
          </th>
          <th
            class="pitchingStatNames"
          >
            ip
          </th>
          <th
            class="pitchingStatNames"
          >
            w
          </th>
          <th
            class="pitchingStatNames"
          >
            gs
          </th>
          <th
            class="pitchingStatNames"
          >
            sv
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tableRowFigs"
        >
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
          <td
            class="figures"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
